### PR TITLE
[train][doc] add elastic training user guide

### PR DESCRIPTION
**Pull request by liulehui**
## Description
1. add user guide for elastic training. 

## Additional information
tested with `make develop && make local`
see below screenshots
image

image

image

### CODE REVIEW

File: doc/source/train/user-guides/elastic-training.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    scaling_config = ScalingConfig(
+        num_workers=(1, 8),
+        use_gpu=True,
+        elastic_resize_monitor_interval_s=30.0,
+    )
+
+Limitations
+-----------
+
+Elastic training is supported for CPU and GPU backends only. It isn't supported yet for TPU training.
```
[the remark this answers is not in the document]
Comment (liulehui) [reply doc/source/train/user-guides/elastic-training.rst:123]: will send an update doc here once I tested elastic training with TPUs!!

File: doc/source/train/user-guides/elastic-training.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+Starting with available workers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Ray Train always requests ``max_workers`` number of workers. If it can't get all of them, it starts when ``min_workers`` is available so training can begin without waiting for the full set of resources.
+
+When failures happen
+~~~~~~~~~~~~~~~~~~~~
+
+If any failures happen (for example, a worker crashes or a node is preempted), Ray Train restarts with fewer workers. It then attempts again to bring the worker group back up to ``max_workers``. Without a retry limit, the run would exit on the first such failure. To allow the run to retry when worker failures occur, configure :attr:`~ray.train.RunConfig.failure_config` with :attr:`~ray.train.FailureConfig.max_failures`:
```
Comment (matthewdeng) [inline doc/source/train/user-guides/elastic-training.rst:45]: Can you also mention setting `max_failures` in the previous section? It seems like it's critical to actually take advantage of elastic training.

File: doc/source/train/user-guides/elastic-training.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    trainer = TorchTrainer(
+        train_func,
+        scaling_config=scaling_config,
+        run_config=run_config,
+    )
+
+When more nodes become available
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If the cluster gets more nodes, Ray Train can resize the worker group. This triggers a graceful restart from the latest checkpoint with the new workers added, so training can use the extra capacity. By default, the controller considers resizing every 60 seconds while the worker group is healthy. To change how often resize decisions are made, set :attr:`~ray.train.ScalingConfig.elastic_resize_monitor_interval_s` in your scaling config:
```
Comment (matthewdeng) [inline doc/source/train/user-guides/elastic-training.rst]: Maybe want to document how to configure cluster autoscaling to match the elastic scaling configuration.

File: doc/source/train/user-guides/elastic-training.rst (hunk 2 above)
Comment (liulehui) [reply doc/source/train/user-guides/elastic-training.rst:45]: done.

File: doc/source/train/user-guides/elastic-training.rst (hunk 3 above)
Comment (liulehui) [reply doc/source/train/user-guides/elastic-training.rst]: added a section below about configuring cluster autoscaling and ref to corresponding doc.

**Review by matthewdeng — APPROVED** [review]
